data/src/main/java/org/apache/iceberg/data/orc/GenericOrcReader.java

```diff
@@ -189,30 +226,89 @@ public Long convert(ColumnVector vector, int row) {
     }
   }
 
-  private static class BinaryConverter implements Converter<byte[]> {
+  private static class TimestampConverter implements Converter<LocalDateTime> {
+    private final ZoneId localZoneId;
+
+    TimestampConverter() {
+      this.localZoneId = ZoneId.systemDefault();
+    }
+
+    private LocalDateTime convert(TimestampColumnVector vector, int row) {
+      return ZonedDateTime
+          .ofInstant(Instant.ofEpochSecond(vector.time[row] / 1_000, vector.nanos[row]), localZoneId)
+          .toLocalDateTime();
+    }
+
+    @Override
+    public LocalDateTime convert(ColumnVector vector, int row) {
+      int rowIndex = vector.isRepeating ? 0 : row;
+      if (!vector.noNulls && vector.isNull[rowIndex]) {
+        return null;
+      } else {
+        return convert((TimestampColumnVector) vector, rowIndex);
+      }
+    }
+  }
+
+  private static class FixedConverter implements Converter<byte[]> {
     @Override
     public byte[] convert(ColumnVector vector, int row) {
       int rowIndex = vector.isRepeating ? 0 : row;
       if (!vector.noNulls && vector.isNull[rowIndex]) {
         return null;
       } else {
         BytesColumnVector bytesVector = (BytesColumnVector) vector;
-        return Arrays.copyOfRange(bytesVector.vector[rowIndex],
-            bytesVector.start[rowIndex],
+        return Arrays.copyOfRange(bytesVector.vector[rowIndex], bytesVector.start[rowIndex],
             bytesVector.start[rowIndex] + bytesVector.length[rowIndex]);
       }
     }
   }
 
+  private static class BinaryConverter implements Converter<ByteBuffer> {
+    @Override
+    public ByteBuffer convert(ColumnVector vector, int row) {
+      int rowIndex = vector.isRepeating ? 0 : row;
+      if (!vector.noNulls && vector.isNull[rowIndex]) {
+        return null;
+      } else {
+        BytesColumnVector bytesVector = (BytesColumnVector) vector;
+        ByteBuffer buf = ByteBuffer.allocate(bytesVector.length[rowIndex]);
+        buf.put(bytesVector.vector[rowIndex], bytesVector.start[rowIndex], bytesVector.length[rowIndex]);
+        buf.rewind();
```

Comment:
This should not copy. Instead, it should point a `ByteBuffer` at the underlying bytes in the vector using `ByteBuffer.wrap(array, offset, length)`.

Reply:
Done